### PR TITLE
Open sound control 2355

### PR DESCRIPTION
**Pull request by pkstone**
OSC enhancements: /tuning/scl, /tuning/kbm, /settings/path/scl, and /settings/path/kbm added. OSC html page updated.

### CODE REVIEW

File: src/common/SurgeStorage.cpp [hunk 1]
```diff
@@ -2269,6 +2269,36 @@ bool SurgeStorage::resetToCurrentScaleAndMapping()
     return true;
 }
 
+void SurgeStorage::loadTuningFromSCL(const fs::path &p)
+{
+    try
+    {
+        retuneToScale(Tunings::readSCLFile({p.u8string()}));
```
Comment (mx) [inline src/common/SurgeStorage.cpp:2276]: I don't think you need the braced initialization here (and below).

File: src/common/SurgeStorage.cpp [hunk 2]
```diff
@@ -2269,6 +2269,36 @@ bool SurgeStorage::resetToCurrentScaleAndMapping()
...
+
+void SurgeStorage::loadMappingFromKBM(const fs::path &p)
+{
+    try
+    {
+        remapToKeyboard(Tunings::readKBMFile({p.u8string()}));
+    }
+    catch (Tunings::TuningError &e)
+    {
+        remapToConcertCKeyboard();
```
Comment (mx) [inline src/common/SurgeStorage.cpp:2295]: Why does the above remap to 12TET on error, and this one to ConcertC? Should probably be consistent.

File: src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp [hunk 3]
```diff
@@ -446,8 +446,10 @@ struct oscParamInfo
 // Sort function for displaying parameters (below)
 bool compareParams(const oscParamInfo &opl, const oscParamInfo &opr)
 {
-    int lcg = opl.p->ctrlgroup;
-    int rcg = opr.p->ctrlgroup;
+    int cgroup_order[10] = {0, 1, 2, 4, 3, 5, 6, 7, 8, 9};
+
+    int lcg = cgroup_order[opl.p->ctrlgroup];
+    int rcg = cgroup_order[opr.p->ctrlgroup];
```
Comment (mx) [inline src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp:452]: I don't understand the necessity of cgroup_order here. It looks like it's mapping 0-9 to 0-9.

File: src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp [hunk 4]
```diff
@@ -652,6 +754,7 @@ code {
                         </table>
                     </div>
                 </div>
+                </div>
```
Comment (mx) [inline src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp:757]: Is this right? It's the same indent level as the one above it.

**Review by mx — CHANGES_REQUESTED** [review]

File: src/common/SurgeStorage.cpp (hunk 2 above)
Comment (pkstone) [reply src/common/SurgeStorage.cpp:2295]: This one is a keyboard mapping (.kbm) which is distinct from a plain tuning (.scl). Its default is 'ConcertCKeyboard()'.

File: src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp (hunk 3 above)
Comment (pkstone) [reply src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp:452]: Four and three are transposed. It was just a way to arrange the control groups on the HTML with a better flow.

File: src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp (hunk 4 above)
Comment (pkstone) [reply src/surge-xt/gui/SurgeGUIEditorHtmlGenerators.cpp:757]: Looks like it's an extra. I removed it.

File: src/common/SurgeStorage.cpp (hunk 1 above)
Comment (pkstone) [reply src/common/SurgeStorage.cpp:2276]: Will fix.

**Review by mx — APPROVED** [review]

**Pull request merged**